### ISSUE TITLE
Support for RGB and Color Temp control of SYLVANIA/LEDVANCE Smart+ Bluetooth bulb

### ISSUE THREAD

**Issue by mj-sakellaropoulos** [Ultraworg/LEDVANCE_BLE_MESH#7]
Hello, Nice work here.

I can confirm it's working for ON/OFF and Brightness control.

What would be required to support control of HSL and Color Temperature elements?
Quick look in [ble mesh feature list](https://docs.espressif.com/projects/esp-idf/en/release-v3.3/api-guides/esp-ble-mesh/ble-mesh-feature-list.html) suggests support for Light HSL and Temp client.

Is this support planned ? If not, I have a working IDF environment for this repo and I could try to add it.

Maybe it's better to try and implement this directly in ESPHome (https://github.com/esphome/feature-requests/issues/1671) ?

**Comment by Bluscream** [comment]
Wait how did you get this working?

**Comment by Ultraworg** [comment]
I unfortunately do not have bulbs with HSL or color temperature so I cannot give a definitive answer.
But if you find these in the nrf mesh app and if you can also control them from there, then it should be possible to also integrate those models.

**Comment by mj-sakellaropoulos** [comment]
> I unfortunately do not have bulbs with HSL or color temperature so I cannot give a definitive answer. But if you find these in the nrf mesh app and if you can also control them from there, then it should be possible to also integrate those models.

Yeah I have possibly found the relevant sections in the ESP-IDP BLE Mesh Stack https://github.com/espressif/esp-idf/blob/a0f798cfc4bbd624aab52b2c194d219e242d80c1/components/bt/esp_ble_mesh/api/models/include/esp_ble_mesh_lighting_model_api.h#L64 , i will give it a shot at some point.

I have also done a small experiement with the `provisioner` example to have the ESP as the mesh provisioner, but for now it is failing to provision the Bulbs (Error reason 0x04 Confirmation Failed).

**Comment by scr4t3r** [comment]
@mj-sakellaropoulos i saw you added HSL support in your fork. however this was before the major code refactoring by @Ultraworg . are you planning on implementing HSL etc also in the refactored codebase?

**Comment by mj-sakellaropoulos** [comment]
> [@mj-sakellaropoulos](https://github.com/mj-sakellaropoulos) i saw you added HSL support in your fork. however this was before the major code refactoring by Ultraworg. are you planning on implementing HSL etc also in the refactored codebase?

Unfortunately, I was never able to get it working. Sending HSL packet would have no effect on the bulb, and crash the Nordic app. From what I remember, the problem is either:
- Some bug or non-standard / proprietary behaviour in ESP, Bulb or Both ble-mesh stacks
- Some bug in my code
- Problem with Old ESP hardware I was using

I may port my fork in the future, although it probably won't happen any time soon.

**Comment by FelixINX** [comment]
I stumble accross this issue and decided to give it a try. I was not able to get it working with ESP, but I had some succes with the nRF Mesh. I have two Sylvania lamps. Here is some information that might help others:

The Temperature control is actually located in the second Element of the light. There is a `Generic Level Server` which can be used to control the temperature directly using the nRF Mesh app. There is also a `Light Ctl Temperature Server` which does not have control from the app, but could possibly be controlled from an ESP.

There is also two others elements : one with a `Light HSL Hue Server` and another with a `Light HSL Saturation Server` (both also have a Generic Level Server, but does not work through the app).

Looking at the [documentation](https://www.bluetooth.com/specifications/specs/basic-lightness-controller-nlc-profile-1-0/) at 3.5, it states:
> The Light CTL Temperature Server model, the Light HSL Hue Server model, and the Light HSL Saturation Server model **should be present on an element with an address higher than the address of the element on which the Light LC Server model is present**.

**Comment by weedy** [comment]
I can get lots of bulbs from local stores, @Ultraworg would you be interested in figuring out more features if I shipped you bulbs?

**Comment by Ultraworg** [comment]
> I can get lots of bulbs from local stores, [@Ultraworg](https://github.com/Ultraworg) would you be interested in figuring out more features if I shipped you bulbs?

@weedy sure, if I could get some lights with color, I could probably try and see what I can find out
